midterm_project/shopping_list_archive.py

```diff
@@ -0,0 +1,125 @@
+import random
+
+annotations = ["-", "*", "+", "-"]
+annotate = random.choice(annotations) * 25
+
+shopping_list_archive = list()
+
+
+class PrintMixin:
+
+    def display_print(self):
+        print("{}\n {}\n{}".format(annotate, self._name, annotate))
+
+        for index, values in enumerate(self._ingredients, start=1):
```

Comment:
if you have ingredients as a property, it is advised to use `self.ingredients` instead of self._ingredients. the `_` attributes are supposed to be protected